File: Illarion.Net/Caster.vb
Imports System.Runtime.InteropServices

<StructLayout(LayoutKind.Explicit)>
Friend Structure Caster
  <FieldOffset(0)>
  Public IntegerValue As Integer
  <FieldOffset(0)>
  Public UIntegerValue As UInteger
  <FieldOffset(0)>
  Public ShortValue1 As Short
  <FieldOffset(2)>
  Public ShortValue2 As Short
  <FieldOffset(0)>
  Public UShortValue1 As UShort
  <FieldOffset(2)>
  Public UShortValue2 As UShort
  <FieldOffset(0)>
  Public ByteValue1 As Byte
  <FieldOffset(1)>
  Public ByteValue2 As Byte
  <FieldOffset(2)>
  Public ByteValue3 As Byte
  <FieldOffset(3)>
  Public ByteValue4 As Byte
End Structure

File: Network/Core/Utility/Caster.cs
using System.Runtime.InteropServices;

namespace Illarion.Network.Core.Utility
{
    [StructLayout(LayoutKind.Explicit)]
    internal struct Caster
    {
        [FieldOffset(0)] public int IntegerValue;
        [FieldOffset(0)] public uint UIntegerValue;
        [FieldOffset(0)] public short ShortValue1;
        [FieldOffset(2)] public short ShortValue2;
        [FieldOffset(0)] public ushort UShortValue1;
        [FieldOffset(2)] public ushort UShortValue2;
        [FieldOffset(0)] public byte ByteValue1;
        [FieldOffset(1)] public byte ByteValue2;
        [FieldOffset(2)] public byte ByteValue3;
        [FieldOffset(3)] public byte ByteValue4;
    }
}